apps/condo/domains/acquiring/schema/RegisterMultiPaymentForVirtualReceiptService.js

```diff
@@ -0,0 +1,215 @@
+    types: [
+        {
+            access: true,
+            type: 'enum RegisterMultiPaymentVirtualReceiptCurrency { RUB, USD }',
+        },
+        {
+            access: true,
+            type: 'input RegisterMultiPaymentVirtualReceiptRecipientInput { bic: String!, bankAccount: String!, accountNumber: String! }',
+        },
+        {
+            access: true,
+            type: 'input RegisterMultiPaymentVirtualReceiptInput { currencyCode: RegisterMultiPaymentVirtualReceiptCurrency!, amount: String!, period: String!, recipient: RegisterMultiPaymentVirtualReceiptRecipientInput! }',
+        },
+        {
+            access: true,
+            type: 'input RegisterMultiPaymentForVirtualReceiptInput { dv: Int!, sender: SenderFieldInput!, receipt: RegisterMultiPaymentVirtualReceiptInput!, acquiringIntegrationContext: AcquiringIntegrationContextWhereUniqueInput! }',
+        },
+        {
+            access: true,
+            type: 'type RegisterMultiPaymentForVirtualReceiptOutput { dv: Int!, multiPaymentId: String!, webViewUrl: String!, feeCalculationUrl: String!, directPaymentUrl: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterMultiPaymentForOneReceipt,
+            schema: 'registerMultiPaymentForVirtualReceipt(data: RegisterMultiPaymentForVirtualReceiptInput!): RegisterMultiPaymentForOneReceiptOutput',
+            resolver: async (parent, args, context) => {
+                // wrap validator function to the current call context
+                const { data } = args
+                const {
+                    sender,
+                    receipt,
+                    acquiringIntegrationContext,
+                } = data
+
+                // Stage 0. Check if input is valid
+                checkDvSender(data, errors.DV_VERSION_MISMATCH, errors.WRONG_SENDER_FORMAT, context)
+
+                // Stage 1: get acquiring context & integration
+                const acquiringContext = await getById('AcquiringIntegrationContext', acquiringIntegrationContext.id)
+
+                if (acquiringContext.deletedAt) {
+                    throw new GQLError(errors.ACQUIRING_INTEGRATION_CONTEXT_IS_DELETED, context)
+                }
+
+                const acquiringIntegration = await AcquiringIntegration.getOne(context, {
+                    id: acquiringContext.integration,
+                })
+
+                if (acquiringIntegration.deletedAt) {
+                    throw new GQLError({
+                        ...errors.ACQUIRING_INTEGRATION_IS_DELETED,
+                        messageInterpolation: { id: acquiringContext.integration },
+                    }, context)
+                }
+
+                // Stage 2. Check VirtualReceipt
+                const { currencyCode, amount, period, recipient: { bic, bankAccount, accountNumber } } = receipt
+
+                // negative to pay value
```

Comment:
I think, that because Amount is a string, literally anything could be here. Like "Hello world"

If we stick wtih `String` I think we should add more validations

Reply:
Yeah, added more validations and tests